### PR TITLE
Check that varargs is applied only to methods

### PR DESCRIPTION
**Pull request by som-snytt**
Check added when typer handles the annotation.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala [hunk 1]
```diff
@@ -1273,6 +1273,9 @@ abstract class RefChecks extends Transform {
         // The getOrElse part should never happen, it's just here as a backstop.
         reporter.error(pos, sym.compileTimeOnlyMessage getOrElse defaultMsg)
       }
+
+      if (!sym.isMethod && sym.hasAnnotation(VarargsClass))
+        reporter.error(sym.pos, s"${sym.name}: Only methods can be marked @varargs.")
```
Comment (hrhino) [inline src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]: I know this is a draught-as-in-beer PR, but this warns when you use the undesired method, not when you declare it.

**Review by hrhino — COMMENTED** [review]
I'm always in favor of more strictening around `@varargs`.

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala (hunk 1 above)
Comment (som-snytt) [reply src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]: Thanks. That saves me from trying it out.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]: I added WIP cream on top.

But now that your name is on it, you're on the hook.

**Comment by som-snytt** [comment]
He never got back to this.

**Comment by som-snytt** [comment]
I wonder if @hrhino is still on the hook to re-review.

**Review by lrytz — APPROVED** [review]

**Pull request merged**